### ISSUE TITLE
Has anybody ever written a gradient for the operator 'SparseReduceSumSparse' ?

### ISSUE THREAD

**Issue by chaihua483** [tensorflow/tensorflow#14964]
I could only find a gradient for `SparseReduceSum`.

```
@ops.RegisterGradient("SparseReduceSum")
def _SparseReduceSumGrad(op, out_grad):
  """Similar to gradient for the Sum Op (i.e. tf.reduce_sum())."""
  sp_indices = op.inputs[0]
  sp_shape = op.inputs[2]
  output_shape_kept_dims = math_ops.reduced_shape(sp_shape, op.inputs[3])
  out_grad_reshaped = array_ops.reshape(out_grad, output_shape_kept_dims)
  scale = sp_shape // math_ops.to_int64(output_shape_kept_dims)
  # (sparse_indices, sparse_values, sparse_shape, reduction_axes)
  return (None, array_ops.gather_nd(out_grad_reshaped, sp_indices // scale),
          None, None)
```

It seems uneasy to write a similar gradient for operator `SparseReduceSumSparse`.
Could anyone give me a hand? Thanks.

**Comment by facaiy** [comment]
Hi, @chaihua483 . `SparseReduceSumSparse` behaves the same way with `SparseReduceSum`, so their gradient implementations are almost same. However, since `gather_nd` doesn't support `SparseTensor`, I cannot find an efficient way to gather values from its `out_grad[1]`.

Above all, I suggest you to use `SparseReduceSum` if you need gradient calculation.

**Comment by chaihua483** [comment]
@facaiy That's right, it might be replaced by `SparseReduceSum ` in small datasets. However, I had to use `SparseReduceSumSparse` with a 3D sparse tensor in my case because it seems still too large if the result 2D matrix is a dense one. I did have implemented a C++ version of the gradient of `SparseReduceSumSparse` and it works well in my code. On the other hand, there exists an operation with no gradient implemented seems not very perfect.

**Comment by facaiy** [comment]
Nice work! However I'm afraid that your new operator is quite like `tf.gather_nd`, right? Do you have an interest in #1950?

**Comment by chaihua483** [comment]
@facaiy Well not exactly, `sparse_tile_like` is much easier rather than `tf.gather_nd` on `SparseTensor`.  Implementing all features included in `tf.gather_nd` seems much harder. 

I would be pleased to have a try when an idea comes to my mind.

**Comment by facaiy** [comment]
cc @suharshs who I think might be interested in the issue. How about implementing the gradient operator in c++ side?

**Comment by jichen-wish** [comment]
+1 for support this

**Comment by wt-huang** [comment]
Closing as this has been resolved.
